src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,400 @@
+end
+
+/-- If the symmetric difference of two negative sets is a null-set, then so are the differences
+between the two sets. -/
+lemma of_diff_eq_zero_of_symm_diff_eq_zero_negative
+  (hu : measurable_set u) (hv : measurable_set v)
+  (hsu : s ≤[u] 0) (hsv : s ≤[v] 0) (hs : s (u Δ v) = 0) :
+  s (u \ v) = 0 ∧ s (v \ u) = 0 :=
+begin
+  rw restrict_le_restrict_iff at hsu hsv,
+  have a := hsu (hu.diff hv) (u.diff_subset v),
+  have b := hsv (hv.diff hu) (v.diff_subset u),
+  erw [of_union (set.disjoint_of_subset_left (u.diff_subset v) set.disjoint_diff)
+        (hu.diff hv) (hv.diff hu)] at hs,
+  rw zero_apply at a b,
+  split,
+  all_goals { linarith <|> apply_instance <|> assumption },
+end
+
+lemma of_inter_eq_of_symm_diff_eq_zero_positive
+  (hu : measurable_set u) (hv : measurable_set v) (hw : measurable_set w)
+  (hsu : 0 ≤[u] s) (hsv : 0 ≤[v] s) (hs : s (u Δ v) = 0) :
+  s (w ∩ u) = s (w ∩ v) :=
+begin
+  have hwuv : s ((w ∩ u) Δ (w ∩ v)) = 0,
+  { refine subset_positive_null_set (hu.union hv) ((hw.inter hu).symm_diff (hw.inter hv))
+      (hu.symm_diff hv) (restrict_le_restrict_union _ _ hu hsu hv hsv) hs _ _,
+    { exact symm_diff_le_sup u v },
+    { rintro x (⟨⟨hxw, hxu⟩, hx⟩ | ⟨⟨hxw, hxv⟩, hx⟩);
+      rw [set.mem_inter_eq, not_and] at hx,
+      { exact or.inl ⟨hxu, hx hxw⟩ },
+      { exact or.inr ⟨hxv, hx hxw⟩ } } },
+  obtain ⟨huv, hvu⟩ := of_diff_eq_zero_of_symm_diff_eq_zero_positive
+    (hw.inter hu) (hw.inter hv)
+    (restrict_le_restrict_subset _ _ hu hsu (w.inter_subset_right u))
+    (restrict_le_restrict_subset _ _ hv hsv (w.inter_subset_right v)) hwuv,
+  rw [← of_diff_of_diff_eq_zero (hw.inter hu) (hw.inter hv) hvu, huv, zero_add]
+end
+
+lemma of_inter_eq_of_symm_diff_eq_zero_negative
+  (hu : measurable_set u) (hv : measurable_set v) (hw : measurable_set w)
+  (hsu : s ≤[u] 0) (hsv : s ≤[v] 0) (hs : s (u Δ v) = 0) :
+  s (w ∩ u) = s (w ∩ v) :=
+begin
+  have hwuv : s ((w ∩ u) Δ (w ∩ v)) = 0,
+  { refine subset_negative_null_set (hu.union hv) ((hw.inter hu).symm_diff (hw.inter hv))
+      (hu.symm_diff hv) (restrict_le_restrict_union _ _ hu hsu hv hsv) hs _ _,
+    { exact symm_diff_le_sup u v },
+    { rintro x (⟨⟨hxw, hxu⟩, hx⟩ | ⟨⟨hxw, hxv⟩, hx⟩);
+      rw [set.mem_inter_eq, not_and] at hx,
+      { exact or.inl ⟨hxu, hx hxw⟩ },
+      { exact or.inr ⟨hxv, hx hxw⟩ } } },
+  obtain ⟨huv, hvu⟩ := of_diff_eq_zero_of_symm_diff_eq_zero_negative
+    (hw.inter hu) (hw.inter hv)
+    (restrict_le_restrict_subset _ _ hu hsu (w.inter_subset_right u))
+    (restrict_le_restrict_subset _ _ hv hsv (w.inter_subset_right v)) hwuv,
+  rw [← of_diff_of_diff_eq_zero (hw.inter hu) (hw.inter hv) hvu, huv, zero_add]
+end
+
+end
+
+end signed_measure
+
+namespace jordan_decomposition
+
+open measure vector_measure signed_measure function
+
+/-- The Jordan decomposition of a signed measure is unique. -/
+theorem to_signed_measure_injective :
+  injective $ @jordan_decomposition.to_signed_measure α _ :=
+begin
+  intros j₁ j₂ hj,
```

Comment:
can you add comments in this proof to explain what you are doing?

Reply:
I was able to shorten the proof a bit while writing the comments :)